Cognite.Simulator.Utils/SimulationRunnerBase.cs

```diff
@@ -368,34 +372,47 @@ async void PublishSimulationRunStatus(ConnectorStatus status, CancellationToken
                 null,
                 token).ConfigureAwait(false);
 
-            SamplingRange samplingRange = null;
-            var validationEnd = startTime;
             var configObj = routineRevision.Configuration;
+            
+            // Determine the validation end time
+            // If the run contains a validation end overwrite, use that instead of the current time
+            var validationEnd = runItem.Run.RunTime.HasValue? CogniteTime.FromUnixTimeMilliseconds(runItem.Run.RunTime.Value) : startTime;
+            
+            SamplingConfiguration samplingConfiguration = null;
             try
             {
-                if (configObj.DataSampling == null)
+                // check if data sampling is enabled
+                if (configObj.DataSampling.Enabled)
                 {
-                    throw new SimulationException($"Data sampling configuration for {routineRevision.ExternalId} missing");
-                }
-                // Determine the validation end time
-                if (runItem.Run.RunTime.HasValue)
+                    // Run validation and return a sampling range
+                    var samplingRange = await SimulationUtils.RunSteadyStateAndLogicalCheck(
+                        _cdfDataPoints,
+                        configObj,
+                        validationEnd,
+                        token).ConfigureAwait(false);
+                    
```

Comment:
@evertoncolling do I understand correctly that SteadyState check and LogicalCheck should only be run if the DataSampling is enabled? Should we in that case forbid enabling these two if DataSampling is disabled? I don't think it happens now on the API validation step

Reply:
Another API change is to make this aggregate field optional when data sampling is disabled, right?
That might be the best to validate in the API as well.

<img width="839" alt="Screenshot 2024-08-15 at 12 24 27" src="https://github.com/user-attachments/assets/7184a6b3-00c0-4897-8175-76a3b9e86d9a">

Was there something else on API side, do you remember?

Reply:
I think you caught everything. In any case let me rewrite it here, to double check:
- Data Sampling Disabled
  - No LC or SSD is performed (they can only be disabled)
  - All datapoints are sampled using latest (before runTime)
  - Aggregates (on timeseries inputs) should be null
- Data Sampling Enabled
  - LC and SSD can be either disabled or enabled
  - All datapoints are sampled using aggregates
  - Sampling Window cannot be null
  - If both LC and SSD are enabled, Validation Window should be null (as it has no effect)
  - Granularity should not be null
  - Aggregates (on timeseries inputs) should not be null